devito/types/grid.py

```diff
@@ -421,3 +425,63 @@ class Interior(SubDomain):
 
     def define(self, dimensions):
         return {d: ('middle', 1, 1) for d in dimensions}
+
+
+class SubDomains(SubDomain):
+    """
+    Class to define a set of N (a positive integer) subdomains.
+
+    Parameters
+    ----------
+    **kwargs
+        * N : int
+            Number of subdomains.
+        * bounds : tuple
+            Tuple of numpy int32 arrays representing the bounds of
+            each subdomain.
+
+    Examples
+    --------
+    Include full example + example of shorthand notation here.
```

Comment:
remainder

Reply:
Reminder? Or is this some comment about Brexit?

Reply:
:joy:  reminder :)